e2e/browser/test-app/components/accessGrants/index.tsx

```diff
@@ -90,34 +93,42 @@ export default function AccessGrant({
     setSharedResourceIri(undefined);
   };
 
-  const handleGrant = async () => {
+  const handleRequest = async () => {
     if (typeof sharedResourceIri !== "string") {
       // If the resource does not exist, do nothing.
       return;
     }
-    const accessGrantRequest = await approveAccessRequest(
-      undefined,
-      {
-        requestor: "https://johndoe.webid",
-        access: { read: true },
-        resources: [sharedResourceIri],
-      },
-      {
-        fetch: session.fetch,
-      },
-    );
-    setAccessGrant(accessGrantRequest);
+    // if the resource owner is not defined, do nothing.
+    if (session.info.webId) {
+      const accessRequestReturned = await issueAccessRequest(
+        {
+          requestor: "https://johndoe.webid",
+          resourceOwner: session.info.webId,
+          access: { read: true },
+          resources: [sharedResourceIri],
+          purpose: [
+            "https://w3id.org/dpv#UserInterfacePersonalisation",
+            "https://w3id.org/dpv#OptimiseUserInterface",
+          ],
+          inherit: false,
+        },
+        {
+          fetch: session.fetch,
+        },
+      );
+      setAccessRequest(accessRequestReturned);
+    }
   };
 
-  const handleRevoke = async () => {
-    if (typeof accessGrant === "undefined") {
+  const handleDenial = async () => {
+    if (typeof accessRequest === "undefined") {
       // If the resource does not exist, do nothing.
       return;
     }
-    await revokeAccessGrant(accessGrant, {
+    await denyAccessRequest(accessRequest, {
       fetch: session.fetch,
     });
-    setAccessGrant(undefined);
+    setAccessRequest(undefined);
```

Comment:
Denying the Access grant is a privileged operation that issues an Access Grant, it has to be performed by an authorized client. This should cancel the Access Request instead.